src/formatDate.js

```diff
@@ -50,7 +50,62 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const inputSeparator = fromFormat[3];
+  const outputSeparator = toFormat[3];
+  const formatDay = 'DD';
+  const formatMonth = 'MM';
+  const shortFormatYear = 'YY';
+  const maxYear = 30;
+  const nineteenthСentury = '19';
+  const twentiethCentury = '20';
+
+  const inputDate = date.split(inputSeparator);
+  let yearInput;
+  let monthInput;
+  let dayInput;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i].includes(shortFormatYear)) {
+      if (inputDate[i].length === shortFormatYear.length) {
+        yearInput = (inputDate[i] < maxYear)
+          ? twentiethCentury + inputDate[i] : nineteenthСentury + inputDate[i];
+      } else {
+        yearInput = inputDate[i];
+      }
+      continue;
+    }
+
+    if (fromFormat[i].includes(formatMonth)) {
+      monthInput = inputDate[i];
+      continue;
+    }
+
+    if (fromFormat[i].includes(formatDay)) {
+      dayInput = inputDate[i];
+    }
+  }
+
+  const outputDateFormat = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i].includes(shortFormatYear)) {
+      if (toFormat[i] === shortFormatYear) {
+        outputDateFormat[i] = yearInput.slice(-2);
+      } else {
```

Comment:
Same situation, consider using `switch` for less nesting and better variable naming